### PR TITLE
Tests for BrowserRouter

### PR DESCRIPTION
**Pull request by als364**
Built a strawman BrowserRouter and guards with nested & sibling ProtectedRoutes and tested all endpoints.

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test:local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/app-router.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+jest.mock('react-router-dom', () => {
+  const originalModule = jest.requireActual('react-router-dom');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    BrowserRouter: ({ children }) => <div>{children}</div>,
+    // useParams: jest.fn(),
+    // useHistory: jest.fn(),
```
Comment (petesantos) [inline ui/src/app/app-router.spec.tsx:15]: Are these comments needed?

**Review by petesantos — APPROVED** [review]
LGTM. 
This is a really nice and easy to follow test - thanks!

File: ui/src/app/app-router.spec.tsx (hunk 1 above)
Comment (als364) [reply ui/src/app/app-router.spec.tsx:15]: No, good catch!

**Comment by als364** [comment]
puppeteer tests timing out on something unrelated to this change. merging

**Pull request merged**